bindings/python/src/models.rs

```diff
@@ -45,28 +46,34 @@ impl PyModel {
 }
 
 impl Model for PyModel {
+    type Trainer = PyTrainer;
+
     fn tokenize(&self, tokens: &str) -> tk::Result<Vec<Token>> {
-        self.model.tokenize(tokens)
+        self.model.read().unwrap().tokenize(tokens)
```

Comment:
I'm always a bit scared by adding that much `unwrap` everywhere... Do you think there's a way we could avoid them ?

Reply:
The `unwrap` here is for the `std::sync::LockResult` that is returned by the `RwLock` when you try to access its content. The `Err` case happens when a thread that was holding the lock panicked. So since this shouldn't happen, and we don't want to recover from it, I think the `unwrap` should be ok.